### ISSUE TITLE
Comment inside XML

### ISSUE THREAD

**Issue by GasDauMin** [end2endzone/ShellAnything#80]
I'am trying to add comment inside the document, at the begining of document and when coment reach some amount of bits it is not loaded at all. 

![image](https://user-images.githubusercontent.com/2140205/103241653-9163fc80-495c-11eb-882e-ad1c92898901.png)

After some time of reviewing, I found a problematic method **Configuration::IsValidConfigFile** it peeks only small portion of configuration file and that portion may not cover important tags like `root`, `shell`, or `menu`.

![image](https://user-images.githubusercontent.com/2140205/103242308-75f9f100-495e-11eb-891c-9c9a1a04d9ae.png)

I thinking maybe it would be better to do check with XML parse tools?

**Comment by end2endzone** [comment]
Outch. I knew that using `--` in xml comments might be bad. I had issues with some software (like visual studio) but since TinyXML (the xml parser used by ShellAnything) was not having issues with these characters, I though it was parser-specific and not something documented in the xml format. Good catch.

The function was first introduced [here](https://github.com/end2endzone/ShellAnything/blob/5aaf05ebb87d2f3f759f63f5f5b88d4fd2ddf73b/src/ConfigManager.cpp#L178) and then it was moved to Configuration class in commit facb33c96e05a144c113f8d099a0f56faf7ee20d and renamed with a capical "I" in commit 2c74f0c0e19a1f77d3e1a43711e676e4a68d9dc7.

In other words: this issue is not linked to a specific issue. I can't remember why I first peeked into the file to know if it was properly formatted instead of directly using an xml parser.
